chalk-solve/src/infer/instantiate.rs

```diff
@@ -103,34 +103,34 @@ impl<I: Interner> InferenceTable<I> {
     }
 }
 
-pub(crate) trait IntoBindersAndValue {
+pub(crate) trait IntoBindersAndValue<'a, I: Interner> {
     type Binders: IntoIterator<Item = ParameterKind<()>>;
     type Value;
 
-    fn into_binders_and_value(self) -> (Self::Binders, Self::Value);
+    fn into_binders_and_value(self, interner: &'a I) -> (Self::Binders, Self::Value);
 }
 
-impl<'a, I, T> IntoBindersAndValue for (&'a Binders<T>, &'a I)
+impl<'a, I, T> IntoBindersAndValue<'a, I> for &'a Binders<T>
 where
     I: Interner,
     T: HasInterner<Interner = I>,
 {
     type Binders = std::iter::Cloned<std::slice::Iter<'a, ParameterKind<()>>>;
     type Value = &'a T;
 
-    fn into_binders_and_value(self) -> (Self::Binders, Self::Value) {
-        (self.0.binders.iter(self.1).cloned(), self.0.skip_binders())
+    fn into_binders_and_value(self, interner: &'a I) -> (Self::Binders, Self::Value) {
+        (self.binders.iter(interner).cloned(), self.skip_binders())
```

Comment:
Oh, I see, it's because we actually return an interior here -- well, this seems ok.